pkg/tsdb/elasticsearch/client/client.go

```diff
@@ -203,6 +202,23 @@ func (c *baseClientImpl) executeRequest(method, uriPath, uriQuery string, body [
 	}, nil
 }
 
+func (c *baseClientImpl) newDatasourceHttpClient() (*http.Client, error) {
```

Comment:
The headers are stored in the http options so we need to create new http options for each request in addition to a client.